File: packages/quran_bridge_ios/ios/Classes/QuranNativeService.swift
import Foundation

/// Bridges method channel calls to native Quran data logic.
final class QuranNativeService {

    let bookmarkStore = BookmarkStore()

    // MARK: - Surahs

    func getSurahs() throws -> [[String: Any]] {
        // TODO: integrate quran-ios library here — replace stub with real data source
        return (1...114).map { number in
            [
                "number": number,
                "name": "Surah \(number)",
                "nameArabic": "",
                "nameTransliteration": "Surah \(number)",
                "ayahCount": 7,
                "revelationType": "meccan",
                "ayahs": [] as [[String: Any]],
            ]
        }
    }

    // MARK: - Pages

    func getPage(_ pageNumber: Int) throws -> [String: Any] {
        guard pageNumber >= 1 && pageNumber <= 604 else {
            throw QuranBridgeError.invalidArgument("pageNumber out of range: \(pageNumber)")
        }
        // TODO: integrate quran-ios library here — fetch real page data
        return [
            "pageNumber": pageNumber,
            "ayahs": [] as [[String: Any]],
            "surahNumbers": [1] as [Int],
            "juzNumber": 1,
        ]
    }

    // MARK: - Verses

    func getVerses(surah: Int, from: Int?, to: Int?) throws -> [[String: Any]] {
        guard surah >= 1 && surah <= 114 else {
            throw QuranBridgeError.invalidArgument("surah out of range: \(surah)")
        }
        // TODO: integrate quran-ios library here — fetch real ayah range
        return []
    }
}

File: packages/quran_bridge_android/android/src/main/kotlin/dev/quran/bridge/QuranNativeService.kt
package dev.quran.bridge

import android.content.Context

/// Bridges method channel calls to native Quran data logic.
class QuranNativeService(private val context: Context) {

    val bookmarkStore = BookmarkStore(context)

    // ── Surahs ───────────────────────────────────────────────────────────────

    fun getSurahs(): List<Map<String, Any>> {
        // TODO: integrate quran_android logic here — replace stub with real data source
        return (1..114).map { number ->
            mapOf(
                "number"              to number,
                "name"                to "Surah $number",
                "nameArabic"          to "",
                "nameTransliteration" to "Surah $number",
                "ayahCount"           to 7,
                "revelationType"      to "meccan",
                "ayahs"               to emptyList<Map<String, Any>>(),
            )
        }
    }

    // ── Pages ────────────────────────────────────────────────────────────────

    fun getPage(pageNumber: Int): Map<String, Any> {
        if (pageNumber < 1 || pageNumber > 604) {
            throw QuranBridgeException.invalidArgs("pageNumber out of range: $pageNumber")
        }
        // TODO: integrate quran_android logic here — fetch real page data
        return mapOf(
            "pageNumber"    to pageNumber,
            "ayahs"         to emptyList<Map<String, Any>>(),
            "surahNumbers"  to listOf(1),
            "juzNumber"     to 1,
        )
    }

    // ── Verses ───────────────────────────────────────────────────────────────

    fun getVerses(surah: Int, from: Int?, to: Int?): List<Map<String, Any>> {
        if (surah < 1 || surah > 114) {
            throw QuranBridgeException.invalidArgs("surah out of range: $surah")
        }
        // TODO: integrate quran_android logic here — fetch real ayah range
        return emptyList()
    }
}
